lib/tapioca/gem/listeners/sorbet_signatures.rb

```diff
@@ -67,7 +67,11 @@ def compile_signature(signature, parameters)
         sig { params(signature: T.untyped).returns(T::Boolean) }
         def signature_final?(signature)
           modules_with_final = T::Private::Methods.instance_variable_get(:@modules_with_final)
-          final_methods = modules_with_final[signature.owner.object_id]
+          final_methods = if sorbet_supports?(:identity_comparison)
+            modules_with_final[signature.owner]
+          else
+            modules_with_final[signature.owner.object_id]
+          end
```

Comment:
Could we not have done:
```ruby
final_methods = modules_with_final[signature.owner] || modules_with_final[signature.owner.object_id]
```

Reply:
Good call!

(For the record, this was fixed in #1734)